### PR TITLE
Add ui to configure text search services

### PR DESCRIPTION
**Pull request by kappu72**
This plugins adds a ui to configure text search services. With this pool configured services will be saved and loaded from map config.
To use,  add  
SearchServicesConfigPlugin: require('../plugins/SearchServicesConfig' to plugins.config and configure in localConfig.json

### CODE REVIEW

File: web/client/components/mapcontrols/searchservicesconfig/SearchServices.css [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+/* This class is applied to the lower origin when
+   its values is > 50%. */
+  z-index: 10;
+}
+.noUi-state-tap .noUi-origin {
+-webkit-transition: left 0.3s, top 0.3s;
+  transition: left 0.3s, top 0.3s;
+}
+.noUi-state-drag * {
+  cursor: inherit !important;
```
Comment (offtherailz) [inline web/client/components/mapcontrols/searchservicesconfig/SearchServices.css]: don't use !important

File: web/client/components/mapcontrols/searchservicesconfig/SearchServices.css [hunk 2]
```diff
@@ -0,0 +1,403 @@
+/* all this css rules are related to the slider, copy paste from advanced layertree example */
+
+.noUi-target,
```
Comment (offtherailz) [inline web/client/components/mapcontrols/searchservicesconfig/SearchServices.css]: these are generic for every noUi, please make them more specific or move the generic css in proper folders

File: web/client/plugins/SearchServicesConfig.jsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
+/**
```
Comment (offtherailz) [inline web/client/plugins/SearchServicesConfig.jsx]: Avoid /** in favor of /* to allow documentation

File: web/client/plugins/SearchServicesConfig.jsx [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+require('../components/mapcontrols/searchservicesconfig/SearchServices.css');
+
+const SearchServicesButton = require('./searchservicesconfig/ToggleButton');
+const ServiceList = require('../components/mapcontrols/searchservicesconfig/ServicesList.jsx');
+const WFSServiceProps = require('../components/mapcontrols/searchservicesconfig/WFSServiceProps.jsx');
+const ResultsProps = require('../components/mapcontrols/searchservicesconfig/ResultsProps.jsx');
+const WFSOptionalProps = require('../components/mapcontrols/searchservicesconfig/WFSOptionalProps.jsx');
+
+
+const SearchServicesConfigPanel = React.createClass({
```
Comment (offtherailz) [inline web/client/plugins/SearchServicesConfig.jsx]: Document the plugin. see http://dev.mapstore2.geo-solutions.it/mapstore/docs/building-and-developing

File: web/client/translations/data.it-IT [hunk 5]
```diff
@@ -266,7 +266,31 @@
             "bearingLabel": "Direzione"
         },
         "search":{
-            "placeholder": "Cerca un indirizzo o inserisci coordinate..."
+            "placeholder": "Cerca un indirizzo o inserisci coordinate...",
```
Comment (offtherailz) [inline web/client/translations/data.it-IT]: for french and german use an online service.

File: web/client/translations/data.en-US [hunk 6]
```diff
@@ -266,7 +266,31 @@
...
+            "configpaneltitle": "Create/edit a search service",
+            "serviceslistlabel": "Available services",
+            "overriedservice": "Overried services",
+            "addbtn": "Add",
+            "nextbtn": "Next",
+            "prevbtn": "Back",
+            "savebtn": "Update",
+            "cancelbtn": "Cancel",
+            "confirmremove": "Delete?",
+            "cancelconfirm": "Confirm Cancel?",
```
Comment (offtherailz) [inline web/client/translations/data.en-US]: `Do you confirm?` or `are you sure?`

File: web/client/translations/data.it-IT [hunk 7]
```diff
@@ -266,7 +266,31 @@
...
+            "searchservicesbutton": "Configura servizi di ricerca",
+            "configpaneltitle": "Aggiungi/modifica un servizio di ricerca",
+            "serviceslistlabel": "Servizi disponibili",
+            "overriedservice": "Sovrascrivi servizi",
+            "addbtn": "Aggiungi",
+            "cancelbtn": "Annulla",
+            "confirmremove": "Elimina?",
+            "cancelconfirm": "Confermi annulla?",
```
Comment (offtherailz) [inline web/client/translations/data.it-IT]: Confermi? or "Sei sicuro?"

File: web/client/translations/data.en-US [hunk 8]
```diff
@@ -266,7 +266,31 @@
             "bearingLabel": "Bearing"
         },
         "search":{
-            "placeholder": "Search by location name or coordinates ..."
+            "placeholder": "Search by location name or coordinates ...",
+            "searchservicesbutton": "Configure search services",
+            "configpaneltitle": "Create/edit a search service",
+            "serviceslistlabel": "Available services",
+            "overriedservice": "Overried services",
```
Comment (offtherailz) [inline web/client/translations/data.en-US]: Override default services

File: web/client/translations/data.it-IT [hunk 9]
```diff
@@ -266,7 +266,31 @@
...
+            "s_priority": "Priorità",
+            "s_url": "Url servizio",
+            "s_layer": "Nome layer",
+            "s_attributes": "Attributi",
+            "s_sort": "Ordina per",
+            "s_max_features": "Max features",
+            "s_wfs_props_label" : "Proprietà WFS service",
+            "s_wfs_opt_props_label" : "Proprietà facoltative",
+            "s_result_props_label": "Proprietà di visualizzazione dei risultati",
+            "s_priority_info": "Usato per oridnare resultati, valori più alti per primi. Nominatim  ha priorità = 5"
```
Comment (offtherailz) [inline web/client/translations/data.it-IT]: `Usato per oridnare resultati,i valori più alti vengono visualizzati per primi. Nominatim ha priorità = 5`
note: only 1 space after "Nominatim"

File: web/client/translations/data.en-US [hunk 10]
```diff
@@ -266,7 +266,31 @@
...
+            "s_priority": "Priority",
+            "s_url": "Service url",
+            "s_layer": "Layer",
+            "s_attributes": "Attributes",
+            "s_sort": "Sort by",
+            "s_max_features": "Max features",
+            "s_wfs_props_label" : "WFS service props",
+            "s_wfs_opt_props_label" : "Optional props",
+            "s_result_props_label": "Result display properties",
+            "s_priority_info": "Used to sort search results, higher values first. Nominatim  results have priority = 5"
```
Comment (offtherailz) [inline web/client/translations/data.en-US]: 1 space only after nominatim

File: web/client/actions/searchconfig.js [hunk 11]
```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2017, GeoSolutions Sas.
```
Comment (MV88) [inline web/client/actions/searchconfig.js]: remove all double * from copyright comments

File: web/client/components/mapcontrols/searchservicesconfig/ResultsProps.jsx [hunk 12]
```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright 2017, GeoSolutions Sas.
```
Comment (MV88) [inline web/client/components/mapcontrols/searchservicesconfig/ResultsProps.jsx]: here

File: web/client/actions/__tests__/searchconfig-test.js [hunk 13]
```diff
@@ -0,0 +1,51 @@
+/**
+* Copyright 2017, GeoSolutions Sas.
```
Comment (MV88) [inline web/client/actions/__tests__/searchconfig-test.js]: here

File: web/client/components/mapcontrols/searchservicesconfig/ServicesList.jsx [hunk 14]
```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2017, GeoSolutions Sas.
```
Comment (MV88) [inline web/client/components/mapcontrols/searchservicesconfig/ServicesList.jsx]: here

File: web/client/components/mapcontrols/searchservicesconfig/WFSOptionalProps.jsx [hunk 15]
```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2017, GeoSolutions Sas.
```
Comment (MV88) [inline web/client/components/mapcontrols/searchservicesconfig/WFSOptionalProps.jsx]: here

File: web/client/components/mapcontrols/searchservicesconfig/WFSServiceProps.jsx [hunk 16]
```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2017, GeoSolutions Sas.
```
Comment (MV88) [inline web/client/components/mapcontrols/searchservicesconfig/WFSServiceProps.jsx]: here

File: web/client/components/mapcontrols/searchservicesconfig/__tests__/SearchServiceConfig-test.jsx [hunk 17]
```diff
@@ -0,0 +1,48 @@
+/**
+* Copyright 2017, GeoSolutions Sas.
```
Comment (MV88) [inline web/client/components/mapcontrols/searchservicesconfig/__tests__/SearchServiceConfig-test.jsx]: here

File: web/client/plugins/searchservicesconfig/ToggleButton.jsx [hunk 18]
```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2017, GeoSolutions Sas.
```
Comment (MV88) [inline web/client/plugins/searchservicesconfig/ToggleButton.jsx]: here

File: web/client/reducers/__tests__/searchconfig-test.js [hunk 19]
```diff
@@ -0,0 +1,78 @@
+/**
+* Copyright 2017, GeoSolutions Sas.
```
Comment (MV88) [inline web/client/reducers/__tests__/searchconfig-test.js]: here

File: web/client/reducers/searchconfig.js [hunk 20]
```diff
@@ -0,0 +1,45 @@
+/**
+* Copyright 2017, GeoSolutions Sas.
```
Comment (MV88) [inline web/client/reducers/searchconfig.js]: here

File: web/client/translations/data.it-IT (hunk 9 above)
Comment (MV88) [inline web/client/translations/data.it-IT]: fix oridnare resultati to ordinare risultati

File: web/client/components/mapcontrols/searchservicesconfig/SearchServices.css [hunk 21]
```diff
@@ -0,0 +1,403 @@
...
+}
+.noUi-marker-vertical.noUi-marker-sub {
+  width: 10px;
+}
+.noUi-marker-vertical.noUi-marker-large {
+  width: 15px;
+}
+
+
+#search-serivices-ui .noUi-background {
```
Comment (offtherailz) [inline web/client/components/mapcontrols/searchservicesconfig/SearchServices.css]: search-serivices-editor is better, and use a class

File: web/client/components/mapcontrols/searchservicesconfig/SearchServices.css [hunk 22]
```diff
@@ -0,0 +1,403 @@
...
+#search-serivices-ui .wfs-required-props-title{
+    margin-bottom: 15px;
+    display: inline-block;
+    margin-left: -10px;
+    margin-right: -10px;
+    width: calc(100% + 20px);
+    border-bottom: 1px solid gray;
+    font-size: larger;
+}
+.confirm-close{
```
Comment (offtherailz) [inline web/client/components/mapcontrols/searchservicesconfig/SearchServices.css]: too generic, use specific selector

File: web/client/plugins/SearchServicesConfig.jsx [hunk 23]
```diff
@@ -0,0 +1,211 @@
...
+            <Portal>
+                <Dialog id={id} style={panelStyle} className={panelClassName}>
+                    <span role="header">
+                        <span>{titleText}</span>
+                        { this.isDirty() ? (
+                            <ConfirmButton className="close" confirming={{
+                                text: <Message msgId="search.cancelbtn" />, className: "btn btn-sm btn-warning confirm-close"}} onConfirm={this.onClose} bsStyle="primary" text={(<Glyphicon glyph={closeGlyph}/>)}/>) : <button onClick={this.onClose} className="close">{closeGlyph ? <Glyphicon glyph={closeGlyph}/> : <span>×</span>}</button>}
+                            }
+                    </span>
+                    <div role="body" id="search-serivices-ui">
```
Comment (offtherailz) [inline web/client/plugins/SearchServicesConfig.jsx]: use a class if possible

**Pull request merged**